docs/how-to-guides/create-a-user-defined-function.md

```diff
@@ -0,0 +1,913 @@
+        String s = (String) map.get("ksql.functions.formula.base.value");
+        baseValue = Integer.parseInt(s);
+    }
+
+    @Udf(description = "The standard version of the formula with integer parameters.")
+    public long formula(@UdfParameter(value = "a") int v1, @UdfParameter(value = "b") int v2) {
+        return (v1 * v2) + baseValue;
+    }
+
+    @Udf(description = "A special variant of the formula, handling double parameters.")
+    public long formula(@UdfParameter(value = "a") double v1, @UdfParameter(value = "b") double v2) {
+        return ((int) (Math.ceil(v1) * Math.ceil(v2))) + baseValue;
+    }
+
+}
+```
+
+Some important points to notice:
+
+
+- The `@UdfDescription` annotation marks the class as a scalar UDF. The `name` parameter gives the function a name so you can refer to it in SQL.
+
+- The `@Udf` annotation marks a method as a body of code to invoke when the function is called. Because ksqlDB is strongly typed, you need to supply multiple signatures if you want your function to work with different column types. This UDF has two signatures: one that takes integer parameters and another that takes doubles.
+
+- The `@UdfParameter` annotation lets you give the function parameters names. These are rendered when using the `DESCRIBE` statement on a function.
+
+- This UDF uses an external parameter, `ksql.functions.formula.base.value`. When a UDF implements the `Configurable` interface, it will be invoked once as the server starts up. `configure()` supplies a map of the parameters that ksqlDB server was started with. You will see how this value is populated later in the guide.
+
+!!! warning
+    External parameters do not yet work for tabular or aggregation functions.
+    Support for these function types will be added soon.
+
+Either continue following this guide by implementing more functions or skip ahead to [compiling the classes](#add-the-uberjar-to-the-classpath) so you can use the functions in ksqlDB.
+
+### Tabular functions
+
+A tabular function (UDTF for short) takes one row as input and produces zero or more rows as output. This is sometimes called "flat map" or "mapcat" in different programming languages. Use this when a value represents many smaller values and needs to be "exploded" into its individual parts to be useful.
+
+Create a file at `src/main/java/com/example/IndexSequenceUdtf.java` and populate it with the following code. This UDTF takes one parameter as input, an array of any type, and returns a sequence of rows, where each element is the element in the array concatenated with its index position as a string.
+
+```java
+package my.example;
+
+import io.confluent.ksql.function.udtf.Udtf;
+import io.confluent.ksql.function.udtf.UdtfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@UdtfDescription(name = "index_seq",
+                 author = "example user",
+                 version = "1.5.0",
```

Comment:
It seems like it might be useful in practice so I figured I'd add it in. I'll call it out the first time it's used so that it doesn't look arbitrary or magical.